spec/rdkafka/bindings_spec.rb

```diff
@@ -100,4 +100,28 @@
       end
     end
   end
+
+  describe "error callback" do
+    context "without an error callback" do
+      it "should do nothing" do
+        expect {
+          Rdkafka::Bindings::ErrorCallback.call(nil, 1, "error", nil)
+        }.not_to raise_error
+      end
+    end
+
+    context "with an error callback" do
+      before do
+        Rdkafka::Config.error_callback = lambda do |error|
+          $received_error = error
+        end
+      end
+
+      it "should call the error callback with an Rdkafka::Error" do
+        Rdkafka::Bindings::ErrorCallback.call(nil, 8, "Broker not available", nil)
+        expect($received_error.code).to eq(:broker_not_available)
```

Comment:
I scratched my head for awhile wondering where this symbol came from before remembering that we call `rd_kafka_err2name` to turn the 8 into `: broker_not_available`. 😆

Reply:
Yeah, I pulled an example out of this: https://github.com/edenhill/librdkafka/blob/58aa3ab832172770b1b690a06900df9c297b135c/src/rdkafka.h#L268-L618